### PR TITLE
PYTHON-4322 Fix racy unified tests of unacknowledged writes

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — APPROVED** [review]

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Hmm I didn't notice the failing test:
```
_ TestCommandLoggingUnacknowledgedWrite.test_An_unacknowledged_write_generates_a_succeeded_log_message_with_ok:_1_reply _

self = <test.test_command_logging.TestCommandLoggingUnacknowledgedWrite testMethod=test_An_unacknowledged_write_generates_a_succeeded_log_message_with_ok:_1_reply>

    def test_case(self):
>       self.run_scenario(spec)

test/unified_format.py:1914: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/unified_format.py:1861: in run_scenario
    self._run_scenario(spec, uri)
test/unified_format.py:1888: in _run_scenario
    self.check_log_messages(spec["operations"], expect_log_messages)
test/unified_format.py:1803: in check_log_messages
    self.assertEqual(len(client["messages"]), len(actual_logs))
E   AssertionError: 2 != 4
```

**Comment by blink1073** [comment]
> Hmm I didn't notice the failing test:

Yeah, I had to update the test runner.

File: test/unified_format.py [hunk 1]
```diff
@@ -1800,7 +1800,8 @@ def format_logs(log_list):
                 clientid = self.entity_map[client["client"]]._topology_settings._topology_id
                 actual_logs = formatted_logs[clientid]
                 actual_logs = [log for log in actual_logs if log["component"] in components]
-                self.assertEqual(len(client["messages"]), len(actual_logs))
+                if not client.get("ignoreExtraMessages", False):
+                    self.assertEqual(len(client["messages"]), len(actual_logs))
```
Comment (ShaneHarvey) [inline test/unified_format.py:1804]: When ignoreExtraMessages is false this new code will mistakenly pass when the test expects messages `[A, B, C]` but actually only sees `[A, B]`. We should handle ignoreExtraMessages by truncating the extras like we do for ignoreExtraEvents:
```python
            if ignore_extra_events:
                actual_events = actual_events[: len(events)]
...
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/unified_format.py (hunk 1 above)
Comment (blink1073) [reply test/unified_format.py:1804]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**